### PR TITLE
Cover LoanSection component by RTL/jest tests

### PR DESCRIPTION
**Pull request by artem-blazhko**
## Purpose

Cover LoanSection component by RTL/jest tests

## Refs

[UICIRC-613](https://issues.folio.org/browse/UICIRC-613)

## Screenshots

100% coverage of component

image

### CODE REVIEW

File: src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+        }
+      });
+    });
+
+    it('"Period" component should get correct properties under "Loan period" section', () => {
+      const expectedProps = {
+        fieldLabel: messageIds.loanPeriod,
+        inputValuePath: 'loansPolicy.period.duration',
+        selectValuePath: 'loansPolicy.period.intervalId',
+        changeFormValue: initialProps.change,
```
Comment (Stanislau-Snapok) [inline src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js:180]: The same with `intervalPeriods` as abowe.
Relates to the whole component, I believe.

File: src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+      };
+
+      expect(Period).toHaveBeenCalledWith(expect.objectContaining(expectedProps), {});
+    });
+
+    it('Should render due date schedule dropdown correctly', () => {
```
Comment (Stanislau-Snapok) [inline src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js:186]: Also we can add `expect` about `schedules`.
You can add `data-testid` for `Field` component and try to find mocked schedules inside.

Comment (artem-blazhko) [reply src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js:186]: It is not necessary to add `data-testid` attribute. I can check visibility of `schedules` without it

Comment (Stanislau-Snapok) [reply src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js:186]: Ofcourse, but with `data-testid` you can do something like:
`expect(within(screen.getByTestId('yourTestId')).getByText('yourScheduleText')).toBeVisible()`
and will be shure that your `schedules` situated inside of correct component.
But it's up to you.

File: src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js (hunk 1 above)
Comment (artem-blazhko) [reply src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js:180]: fixed

File: src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+      expect(Field).toHaveBeenNthCalledWith(4, expect.objectContaining(expectedProps), {});
+      expect(container.getByText(messageIds.itemLimit)).toBeInTheDocument();
+    });
+
+    it('"Period" component should get correct properties under "Grace period" section', () => {
+      const expectedProps = {
+        fieldLabel: messageIds.gracePeriod,
+        inputValuePath: 'loansPolicy.gracePeriod.duration',
+        selectValuePath: 'loansPolicy.gracePeriod.intervalId',
+        changeFormValue: initialProps.change,
```
[the remark this answers is not in the document]
Comment (artem-blazhko) [reply src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js:155]: fixed

File: src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js (hunk 2 above)
Comment (artem-blazhko) [reply src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js:186]: checked as children property instead

File: src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+    });
+
+    it('Should render loan profile dropdown correctly', () => {
+      const expectedProps = {
+        component: Select,
+        id: 'input_loan_profile',
+        name: 'loansPolicy.profileId',
+      };
+
+      expect(Field).toHaveBeenNthCalledWith(2, expect.objectContaining(expectedProps), {});
```
[the remark this answers is not in the document]
Comment (artem-blazhko) [reply src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js:120]: fixed

**Review by Stanislau-Snapok — APPROVED** [review]

File: src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+  describe('Updating of component', () => {
+    let container;
+
+    beforeEach(() => {
+      container = renderLoansSection();
+    });
+
+    it('Should not trigger "change" property after component properties updating', () => {
+      container.rerender(<LoansSection {...initialProps} schedules={[]} />);
+      expect(initialProps.change).not.toBeCalled();
```
Comment (pavelspichonak) [inline src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js:316]: Not sure but it looks like we need to clear this mock `initialProps.change` where we clear `Field` component mock

**Review by pavelspichonak — APPROVED** [review]

File: src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js (hunk 5 above)
Comment (artem-blazhko) [reply src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js:316]: done

**Pull request merged**